ocean_lib/data_provider/test/test_data_service_provider.py

```diff
@@ -173,6 +181,51 @@ def test_encrypt(web3, provider_wallet):
     assert decrypted_document_string == test_string
 
 
+def test_initialize(web3, config, publisher_wallet, publisher_ocean_instance):
+    erc721_token, erc20_token = deploy_erc721_erc20(
+        web3, config, publisher_wallet, publisher_wallet
+    )
+    metadata = {
+        "created": "2020-11-15T12:27:48Z",
+        "updated": "2021-05-17T21:58:02Z",
+        "description": "Sample description",
+        "name": "Sample asset",
+        "type": "dataset",
+        "author": "OPF",
+        "license": "https://market.oceanprotocol.com/terms",
+    }
+    file1_dict = {"type": "url", "url": "https://url.com/file1.csv", "method": "GET"}
+    file2_dict = {"type": "url", "url": "https://url.com/file2.csv", "method": "GET"}
+    file1 = FilesTypeFactory(file1_dict)
+    file2 = FilesTypeFactory(file2_dict)
+
+    # Encrypt file objects
+    encrypt_response = DataSP.encrypt(
+        [file1, file2], "http://localhost:8030/api/services/encrypt"
+    )
+    encrypted_files = encrypt_response.content.decode("utf-8")
+    ddo = publisher_ocean_instance.assets.create(
+        metadata=metadata,
+        publisher_wallet=publisher_wallet,
+        encrypted_files=encrypted_files,
+        erc721_address=erc721_token.address,
+        deployed_erc20_tokens=[erc20_token],
+    )
+    access_service = ddo.get_service(ServiceTypes.ASSET_ACCESS)
+
+    initialize_result = DataSP.initialize(
+        did=ddo.did,
+        service_id=access_service.id,
+        file_index=0,
+        consumer_address=publisher_wallet.address,
+        service_endpoint="http://172.15.0.4:8030/api/services/initialize",
+    )
+    assert initialize_result
+    assert initialize_result.status_code == 200
+    response_json = initialize_result.json()
+    assert response_json["providerFee"] == get_provider_fees()
```

Comment:
this is good. The providerFees returned from our test provider will probably always correspond to our test version. If this changes, we'll adapt the test accordingly. Leave it as is for now.